src/ctap2/pin.rs

```diff
@@ -208,9 +236,9 @@ impl<'a, T: TrussedRequirements> PinProtocol<'a, T> {
 
     // in spec: verify(pinUvAuthToken, ...)
     pub fn verify_pin_token(&mut self, data: &[u8], signature: &[u8]) -> Result<&PinToken> {
-        let pin_token = self.pin_token();
+        let pin_token = self.pin_token().ok_or(Error::PinAuthInvalid)?;
         if pin_token.state.is_in_use && self.verify(pin_token.key_id, data, signature) {
-            Ok(self.pin_token())
+            Ok(self.pin_token().unwrap())
```

Comment:
I would prefer to avoid panics in this case.

Reply:
Added a comment.